src/main/java/com/google/impactdashboard/server/api_utilities/RecommendationRetriever.java

```diff
@@ -20,14 +33,62 @@ private RecommendationRetriever(RecommenderClient recommender) {
    * @return A new instance of a {@code RecommendationRetriever}
    */
   public static RecommendationRetriever create() throws IOException {
-    return new RecommendationRetriever(RecommenderClient.create());
+    RecommenderStubSettings stub = RecommenderStubSettings.newBuilder()
+        .setCredentialsProvider(Credentials::getCredentials).build();
+    return new RecommendationRetriever(RecommenderClient.create(RecommenderSettings.create(stub)));
   }
+
   /**
    * Helper method to be called to retrieve all recommendations in the last 90 days
    * from recommender for a certain project.
    * @return collection of recommendations for the project specified
    */
-  public Collection<Recommendation> listRecommendations(String projectId) {
-   throw new UnsupportedOperationException("Not implemented");
+  public List<Recommendation> listRecommendations(Collection<LogEntry> recommendationLogs,
+                                String projectId, Recommendation.RecommenderType type,
+                                IamBindingRetriever iamRetriever) {
+    return recommendationLogs.stream().map(recommendationLog -> {
+      Map<String, Value> recommendationDataMap = recommendationLog.getJsonPayload().getFieldsMap();
+      com.google.cloud.recommender.v1.Recommendation recommendation = recommender
+          .getRecommendation(recommendationDataMap.get("recommendationName").getStringValue());
+      List<RecommendationAction> actions = getRecommendationActions(recommendation);
+      return Recommendation.create(projectId, recommendationDataMap.get("actor").getStringValue(),
+          actions, type, recommendationLog.getTimestamp().getSeconds() * 1000,
+          IAMRecommenderMetadata.create(iamRetriever.getActionImpact(actions)));
+    }).collect(Collectors.toList());
+  }
+
+  /**
+   * Takes the information in a recommendation and compiles it into an Action list.
+   * @param recommendation the recommendation the description is coming from
+   * @return The a list of actions for the given recommendation
+   */
+  private List<RecommendationAction> getRecommendationActions(
+      com.google.cloud.recommender.v1.Recommendation recommendation) {
+    RecommendationAction.ActionType type;
+    if(recommendation.getRecommenderSubtype().equals("REMOVE_ROLE")) {
+      type = RecommendationAction.ActionType.REMOVE_ROLE;
+    } else {
+      type = RecommendationAction.ActionType.REPLACE_ROLE;
+    }
+    List<RecommendationAction> actions = recommendation.getContent().getOperationGroupsList()
+        .stream().map(operationGroup -> {
+          AtomicReference<String> affectedAccount = new AtomicReference<>();
+          AtomicReference<String> previousRole = new AtomicReference<>();
+          AtomicReference<String> newRole = new AtomicReference<>("");
```

Comment:
Likewise, it looks like this isn't really multi threaded, is this necessary?

Reply:
The reason we use it here is because these variables is because they get referenced in the foreach and outside the for each since they are changed based on which operation is happening. (i.e newRole is changed with an add operation previousRole with remove).

Reply:
Fixed